File: ModuleCommon/src/main/java/com/qihuan/commonmodule/base/BaseActivity.java
package com.qihuan.commonmodule.base;

import android.support.v7.app.ActionBar;
import android.support.v7.app.AlertDialog;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.Toolbar;
import android.view.View;

import com.qihuan.commonmodule.R;
import com.qihuan.commonmodule.bus.BindEventBus;

import org.greenrobot.eventbus.EventBus;

/**
 * BaseActivity
 *
 * @author Qi
 * @date 2017/6/20
 */

public abstract class BaseActivity extends AppCompatActivity {

    private AlertDialog progressDialog;

    @Override
    protected void onStart() {
        super.onStart();
        if (this.getClass().isAnnotationPresent(BindEventBus.class)) {
            EventBus.getDefault().register(this);
        }
    }

    @Override
    protected void onStop() {
        super.onStop();
        if (this.getClass().isAnnotationPresent(BindEventBus.class)) {
            EventBus.getDefault().unregister(this);
        }
    }

    protected void setToolBar(Toolbar toolbar, String title) {
        toolbar.setTitle(title);
        setSupportActionBar(toolbar);
        ActionBar bar = getSupportActionBar();
        if (bar != null) {
            bar.setDisplayHomeAsUpEnabled(true);
            bar.setDisplayShowHomeEnabled(true);
        }
        toolbar.setNavigationOnClickListener(view -> onBackPressed());
    }


    public void showProgressLoading() {
        if (progressDialog == null) {
            progressDialog = new AlertDialog.Builder(this, R.style.ProgressDialog).create();
        }
        View loadView = getLayoutInflater().inflate(R.layout.alert_progress, null);
        progressDialog.setView(loadView, 0, 0, 0, 0);
        progressDialog.setCanceledOnTouchOutside(false);
        if (progressDialog.isShowing()) {
            return;
        }
        progressDialog.show();
    }

    public void hideProgressLoading() {
        if (progressDialog != null && progressDialog.isShowing()) {
            progressDialog.dismiss();
        }
    }
}

File: ModuleCommon/src/main/java/com/qihuan/commonmodule/base/BaseActivity.kt
package com.qihuan.commonmodule.base

import android.graphics.Color
import android.os.Bundle
import android.support.annotation.ColorInt
import android.support.v7.app.AppCompatActivity
import com.gyf.barlibrary.ImmersionBar
import com.qihuan.commonmodule.bus.BindEventBus
import com.qihuan.commonmodule.views.LoadingDialog
import com.qihuan.commonmodule.views.TitleBar
import org.greenrobot.eventbus.EventBus

/**
 * BaseActivity
 *
 * @author Qi
 * @date 2017/6/20
 */

abstract class BaseActivity : AppCompatActivity(), LoadingDialogManager {

    override val loadingDialog: LoadingDialog by lazy { LoadingDialog(this) }
    protected lateinit var mStatusBar: ImmersionBar

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        mStatusBar = ImmersionBar.with(this)
        mStatusBar.init()
    }

    override fun onDestroy() {
        super.onDestroy()
        mStatusBar.destroy()
        loadingDialog.dismiss()
    }

    override fun onStart() {
        super.onStart()
        if (this.javaClass.isAnnotationPresent(BindEventBus::class.java)) {
            EventBus.getDefault().register(this)
        }
    }

    override fun onStop() {
        super.onStop()
        if (this.javaClass.isAnnotationPresent(BindEventBus::class.java)) {
            EventBus.getDefault().unregister(this)
        }
    }

    protected fun setToolBar(toolbar: TitleBar, title: String = "", @ColorInt color: Int = Color.parseColor("#00000000")) {
        toolbar.setOnBackClickListener { onBackPressed() }
        toolbar.setTitleText(title)
        toolbar.setBackgroundColor(color)
    }
}
